README.md

```diff
@@ -279,16 +282,35 @@ The content viewset usually doesn't require any additinal routes, so you can lea
 
 ## Remote
 
-A remote knows specifics of the plugin Content to put it into Pulp. Remote defines how to synchronize remote content.
-Pulp Platform provides support for concurrent downloading of remote content. Plugin writer is encouraged to use one of
-them but is not required to.
+Remotes provide metadata about how content should be downloaded into Pulp, such as the URL of the remote source, the download policy, and some authentication settings. The base ``Remote`` class provided by Pulp Platform provides support for concurrent downloading of remote content. Plugin writers are encouraged to use one of them but is not required to.
 
 ### Model
 
 First model your remote. This file is located at [{{ plugin_snake }}/app/models.py]({{ plugin_snake }}/app/models.py).
 Add any fields that correspond to the remote source.
 
-Remember to define the ``TYPE`` class attribute which is used for filtering purposes,
+Remember to define the ``TYPE`` class attribute which is used for filtering purposes.
+
+### Serializer
+
+Next, add a corresponding serializer field on the in [{{ plugin_snake }}/app/serializers.py]({{ plugin_snake }}/app/serializers.py).
+
+### Viewset
+
+Last, add any additional routes to your [{{ plugin_snake }}/app/viewsets.py]({{ plugin_snake }}/app/viewsets.py).
+The remote viewset usually doesn't require any additinal routes, so you can leave this alone for now.
+
+## Repository
+
+A Repository knows specifics of which Content it supports and defines how to create new RepositoryVersions, and how to validate that those RepositoryVersions are valid.
```

Comment:
Can this be line-split to 79 or 100 chars. Also the third part of the sentence makes it a run on; probably make that its own sentence.